torch/utils/_sympy/printers.py

```diff
@@ -333,6 +337,10 @@ def _print_IntTrueDiv(self, expr: sympy.Expr) -> str:
     # TODO: PowByNatural: we need to implement our own int-int pow.  Do NOT
     # use std::pow, that operates on floats
     def _print_PowByNatural(self, expr: sympy.Expr) -> str:
+        # Implement the special-case of 2**x for now
+        base, exp = expr.args
+        if base == 2:
+            return f"(1 << ({self._print(exp)}))"
```

Comment:
Can exp be a non-integer or negative here?

Reply:
No, they are non-negative integers.

Reply:
But can they be a float for other models? I see a _print_ToFloat method in this class

Reply:
No `PowByNatural` is only for exponents that are natural numbers (non-negative integers).